allow github ci ratelimiting on installer scripts to be managed with UV_GITHUB_TOKEN

### Summary

```
curl -LsSf https://astral.sh/uv/0.6.14/install.sh | sh
downloading uv 0.6.14 x86_64-unknown-linux-gnu
curl: (22) The requested URL returned error: 403
failed to download https://github.com/astral-sh/uv/releases/download/0.6.14/uv-x86_64-unknown-linux-gnu.tar.gz
this may be a standard network error, but it may also indicate
that uv's release process is not working. When in doubt
please feel free to open an issue!
make: *** [Makefile:12: install-uv] Error 1

Exited with code exit status 2
```

Some of our CI's are encoutering the following error from time to time.

I'm not sure but i guess we are being rate limited by Github

Is there a way to authenticate the installation call ?

### Platform

Ubuntu 22.04

### Version

uv 0.6.14

### Python version

python 3.11

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -40,6 +40,7 @@ permissions:
 # If there's a prerelease-style suffix to the version, then the release(s)
 # will be marked as a prerelease.
 on:
+  pull_request:
   workflow_dispatch:
     inputs:
       tag:
@@ -68,7 +69,7 @@ jobs:
         # we specify bash to get pipefail; it guards against the `curl` command
         # failing. otherwise `sh` won't catch that `curl` returned non-0
         shell: bash
-        run: "curl --proto '=https' --tlsv1.2 -LsSf https://github.com/astral-sh/cargo-dist/releases/download/v0.28.4-prerelease.1/cargo-dist-installer.sh | sh"
+        run: "curl --proto '=https' --tlsv1.2 -LsSf https://github.com/astral-sh/cargo-dist/releases/download/v0.28.4/cargo-dist-installer.sh | sh"
       - name: Cache dist
         uses: actions/upload-artifact@6027e3dd177782cd8ab9af838c04fd81a07f1d47
         with:

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -294,7 +294,7 @@ inherits = "release"
 # Config for 'dist'
 [workspace.metadata.dist]
 # The preferred dist version to use in CI (Cargo.toml SemVer syntax)
-cargo-dist-version = "0.28.4-prerelease.1"
+cargo-dist-version = "0.28.4"
 # make a package being included in our releases opt-in instead of opt-out
 dist = false
 # CI backends to support
@@ -330,7 +330,7 @@ auto-includes = false
 # Whether dist should create a Github Release or use an existing draft
 create-release = true
 # Which actions to run on pull requests
-pr-run-mode = "skip"
+pr-run-mode = "plan"
 # Whether CI should trigger releases with dispatches instead of tag pushes
 dispatch-releases = true
 # Which phase dist should use to create the GitHub release

